initfiles/etc/DIR_NAME/environment.xml.in

```diff
@@ -15,10 +15,9 @@
   <runtime>${RUNTIME_PATH}</runtime>
   <sourcedir>${CONFIG_SOURCE_PATH}</sourcedir>
   <user>${RUNTIME_USER}</user>
-  <ports>
```

Comment:
It doesn't appear that CProperties in jlib supports creating nested properties, and that's how that sections is written/read by the config system. Also, https://track.hpccsystems.com/browse/HPCC-12750?jql=text%20~%20%22generate%20environment%22 would completely remove the static environment.xml from the source tree, which would eliminate the nesting entirely from source, but retain backward compatibility.  

If we need hierarchy in the EnvSettings section of the environment.xml we could convert it to XSD based, which probably is the right thing to do long term for every config setting.